### DISCUSSION TITLE
Monitoring HTTP Requests Transactions

### DISCUSSION THREAD

**Discussion by douglaslins11** [valhalla/valhalla#4265] [Q&A] [answered]
Hello everybody

I'm deploying Valhalla here in my environment and I currently need to instrument it to collect some metrics, initially I want to specify the beginning and end of the http requests, but I didn't find the entrypoint of the requests in the code for me to call my functions that start and end a transaction. Can you help me with this?

**Comment by kevinkreiser** [comment, marked as answer]
if you are using statsd valhalla supports that already. valhalla/worker.cc will have some clues. if you use some other framework you probably still want to look at where we do statsd stuff. but if you want to just know the main entry point it is here: https://github.com/valhalla/valhalla/blob/master/src/loki/worker.cc#L326

**Comment by douglaslins11** [comment]
@kevinkreiser thanks for the feedback.
I saw that Thor has the same method, what is the difference between the 2?
https://github.com/valhalla/valhalla/blob/master/src/thor/worker.cc#L116

**Reply by nilsnolde** [reply]
Different workers with different jobs. README and docs will help.

**Reply by douglaslins11** [reply]
@nilsnolde which location in the README makes this difference evident?

**Reply by nilsnolde** [reply]
https://github.com/valhalla/valhalla#organization
